src/ssort/_main.py

```diff
@@ -123,6 +127,10 @@ def _on_wildcard_import(**kwargs):
                 )
             else:
                 sys.stderr.write(f"Sorting {str(path)!r}\n")
+                if newline != "\n":
```

Comment:
The duplication of this logic in both `_main` and `_ssort` is a bit confusing to read. Maybe not in this PR, but in the future, I think it would be best if we could consolidate this in some way.

Reply:
I agree.  I had an experiment with trying to consolidate some of the logic into a `Codec` class with `encode` and `decode` methods, but most of the exception handling code remained duplicated.  Will leave as is for now.